Documentation/development-flow.md

```diff
@@ -239,9 +239,40 @@ go tool cover -html=coverage.out -o coverage.html
 ```
 
 #### Running the Integration Tests
+
 For instructions on how to execute the end to end smoke test suite,
 follow the [test instructions](https://github.com/rook/rook/blob/master/tests/README.md).
 
+### Commit structure
+
+Rook maintainers value clear, lengthy and explanatory commit messages. So by default each of your commits must:
+
+* be prefixed by the component it's affecting, if Ceph, then the title of the commit message should be `ceph: my commit title`. If not the commit-lint bot will complain.
+* contain a commit message which explains the original issue and how it was fixed if a bug.
+If a feature it is a full description of the new functionality.
+* refer to the issue it's closing, this is mandatory when fixing a bug
+* have a sign-off, this is achieved by adding `-s` when committing so in practice run `git commit -s`. If not the DCO bot will complain.
+If you forgot to add the sign-off you can also amend a previous commit with the sign-off by running `git commit --amend -s`.
+If you've pushed your changes to Github already you'll need to force push your branch with `git push -f`.
+
+Here is an example of an acceptable commit message:
+
+```text
+component: commit title
+
+This is the commit message, here I'm explaining, what the bug was along with its root cause.
+Then I'm explaining how I fixed it.
+
+Closes: https://github.com/rook/rook/issues/<NUMBER>
+Signed-off-by: First Name Last Name <email address>
```

Comment:
Shouldn't there be a new line between the last line and the sign-off?

Reply:
Hum I think it's done by git automatically.

Reply:
Sorry I meant a new line between the closes and sign-off, like this:
```
Closes: ...

Signed-off-by:...
```

Reply:
Oh I see, I've never done that and I don't think it's really aesthetic :-)